SpatialGDK/Source/SpatialGDK/Private/Tests/RPCContainerTest.cpp

```diff
@@ -0,0 +1,232 @@
+	{
+		return FUnrealObjectRef{ Worker_EntityId(TargetObject), 0 };
+	}
+
+	static uint32 GeneratePayloadFunctionIndex()
+	{
+		static uint32 freeIndex = 0;
+		return freeIndex++;
+	}
+
+	static FPendingRPCParamsPtr CreateMockParameters(UObject* TargetObject, ESchemaComponentType Type)
+	{
+		// Use PayloadData as a place to store RPC type
+		RPCPayload Payload(0, GeneratePayloadFunctionIndex(), TypeToArray(Type));
+		int ReliableRPCIndex = 0;
+
+		FUnrealObjectRef ObjectRef = GenerateObjectRef(TargetObject);
+
+		return MakeUnique<FPendingRPCParams>(ObjectRef, MoveTemp(Payload), ReliableRPCIndex);
+	}
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_nothing_has_been_added_THEN_nothing_is_queued)
+{
+	UMockObject* TargetObject = NewObject<UMockObject>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr Params = CreateMockParameters(TargetObject, ReliableType);
+	FRPCContainer RPCs;
+
+	TestFalse("No queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, ReliableType));
+
+    return true;
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_one_value_has_been_added_THEN_it_is_queued)
+{
+	UMockObject* TargetObject = NewObject<UMockObject>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr Params = CreateMockParameters(TargetObject, ReliableType);
+	FRPCContainer RPCs;
+
+	RPCs.QueueRPC(MoveTemp(Params), ReliableType);
+
+	TestTrue("Queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, ReliableType));
+
+    return true;
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_two_values_of_same_type_have_been_added_THEN_they_are_queued)
+{
+	UMockObject* TargetObject = NewObject<UMockObject>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr Params1 = CreateMockParameters(TargetObject, UnreliableType);
+	FPendingRPCParamsPtr Params2 = CreateMockParameters(TargetObject, UnreliableType);
+	FRPCContainer RPCs;
+
+	RPCs.QueueRPC(MoveTemp(Params1), UnreliableType);
+	RPCs.QueueRPC(MoveTemp(Params2), UnreliableType);
+
+	TestTrue("Queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, UnreliableType));
+
+    return true;
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_one_value_has_been_added_and_processed_THEN_nothing_is_queued)
+{
+	UMockObject* TargetObject = NewObject<UMockObject>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr Params = CreateMockParameters(TargetObject, ReliableType);
+	FRPCContainer RPCs;
+
+	RPCs.QueueRPC(MoveTemp(Params), ReliableType);
+
+	FProcessRPCDelegate Delegate;
+	Delegate.BindUObject(TargetObject, &UMockObject::ProcessRPC);
+	RPCs.ProcessRPCs(Delegate);
+
+	TestFalse("No queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, ReliableType));
+
+    return true;
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_two_values_of_same_type_have_been_added_and_processed_THEN_nothing_is_queued)
```

Comment:
GIVEN_a_container_storing_two_values_of_same_type_WHEN_processed_once_THEN_nothing_is_queued ?

Reply:
This test checks that after the call to ProcessRPCs the container is empty.

Reply:
I think the name I've suggested places more of a distinction between the precondition and action. This, I think, makes it easier for the reader to discern the intent of the system (when you read several tests to get a picture of what the container does). It's also significant, I think, that one call to process is required.

Reply:
Updated test names.